x-pack/plugins/enterprise_search/public/applications/app_search/components/result/types.ts

```diff
@@ -33,3 +33,10 @@ export type Result = {
   // You'll need to cast it to FieldValue whenever you use it.
   [key: string]: object;
 };
+
+export interface ResultAction {
+  onClick(): void;
+  title: string;
+  iconType: string;
+  iconColor?: string;
+}
```

Comment:
Feel free to shout if you're not a fan of the API, in theory I could make it more permissive (passing blanket props?) or closer to EuiButtonIcon, but figured I'd make it simple for now. We can also def tweak this later (off the top of my head Curations is the main place we _have_ custom actions, but maybe I've forgotten somewhere else in the app)